gdk/common/utils.py

```diff
@@ -165,14 +165,37 @@ def parse_json_error(err):
     msg = err.msg
 
     for err_msg, causes in syntax_error_message.JSON_LIBRARY_ERROR_MESSAGES.items():
-        if msg in err_msg:
+        if err_msg in msg:
             for cause in causes:
                 logging.info("\t " + cause)
             break
 
     logging.info("If none of the above is the cause, please review the overall JSON syntax and resolve any issues.")
 
 
+def parse_yaml_error(err):
+    """
+    Log and display a YAML syntax error message.
+
+    Parameters
+    ----------
+    err : yaml.YAMLError
+        The YAML syntax error.
+
+    """
+    logging.error(f"{err}")
+    logging.info("This might be caused by one of the following reasons: ")
+    msg = err.problem
+
+    for err_msg, causes in syntax_error_message.YAML_LIBRARY_ERROR_MESSAGES.items():
+        if err_msg in msg:
```

Comment:
What if the newer version of `yaml` library gives us a new error message that isn't part of the list we're maintaining? In that case, we don't want the  following info log.

Reply:
It's quite uncommon for the yaml library to alter its error messages. However, if a new error message does arise that isn't on our maintained list, the info log won't include potential causes for that specific message. Rest assured, the error message itself will still be displayed, allowing developers to identify and troubleshoot the issue effectively.

Reply:
I was referring to the info log in L196.

Reply:
I can reposition the statement "If none of the above is the cause" within the if clause so that it's displayed only when potential causes are logged.